doc/coding-conventions.xml

```diff
@@ -56,25 +56,30 @@ foo { arg = ...; }
      or list elements should be aligned:
 <programlisting>
 # A long list.
-list =
-  [ elem1
-    elem2
-    elem3
-  ];
+list = [
+  elem1
+  elem2
+  elem3
+];
```

Comment:
What would you suggest for this common case?

```nix
list =
  [
    elem1
    elem2
    elem4
  ] ++ lib.optionals stdenv.isDarwin [ elem4 elem5 ];
```

I like the extra indentation, since it needs top-level keys easy to spot, but if the `[` is on the same line as the `=`, that doesn't really make sense.

Reply:
```nix
{
  list = [
    elem1
    elem2
    elem3
  ] ++ lib.optionals stdenv.isDarwin [
    elem4
    elem5
  ];
}
```

looks good to me, though it is kind of similar to

```
someParameter = [ {
    foo = bar;
    baz = quux;
  } {
    some = more;
    stuff = here;
} ];
```

Reply:
I think your version is fine.

I only really care about the mechanics of what happens when maintaining the code. So when the code evolves, if `elem{1,2,3}` are on distinct lines with no `[]` they are easily movable / commentable / sortable. A soft rule would be that if there are more than 2 elements in a list it's better to use the multi-line notation.

A variation of your example I have seen which is also quite nice is:
```
{
  list =
    [
      elem1
      elem2
      elem3
    ]
    ++ lib.optionals stdenv.isDarwin [ elem4 elem5 ]
    ++ lib.optionals stdenv.isLinux [ elem6 ]
    ;
}
```

Again here the same logic applies: each of the extra `++` is easily movable without having to worry of moving the `;` around.